### PR TITLE
Record landing page and form improvements

### LINKED ISSUE #1623: schema validation: make sure the UI presentation keys are valid
The UI uses a 'presentation' field declared in the schemas that should contain a list of properties to be displayed, in order, e.g.:

````
        "b2share": {
            "presentation": {
                "major": [ "community", "titles", "descriptions", "creators", "open_access", "embargo_date", "license", "disciplines", "keywords", "contact_email" ],
                "minor": [ "contributors", "resource_types", "alternate_identifiers", "version", "publisher", "language"]
            }
        }
````

During schema validation, we should check the presentation items and make sure all the ids are present in the actual schema.

### LINKED ISSUE #1683: ui: deletion of dropdown input for community specific metadata not working
When one creates multiple entries of 'enum'-fields in community specific metadata elements with B2SHARE UI, deletion of entries doesn't work as it should.

- Create three elements
- Press delete on the 1st or 2nd one -> 3rd one gets removed

### LINKED ISSUE #1692: UI doesn't allow deletion of some metadata fields
Record is still a draft phase

**User has inputted metadata to e.g. 'Contributors', but wants to remove the inputted value.**
**However, there is no way or button to remove an entry from 'Contributors'.**

Maybe this is related to the fact that 'Contributors' is defined as type 'object' in JSON Schema?

Nevertheless, this is a bug in the B2SHARE UI

### LINKED ISSUE #1726: UI: Display the DOI a record will get when it is published
Display the DOI a record will get when it is published in the UI.
ie. "this is the DOI this draft will get when it is published as a record."

DOI the record gets is by schema "<DOI_PREFIX>/b2share.<RECORD_UUID>".
- "<DOI_PREFIX>" is configurable by config.py (https://github.com/EUDAT-B2SHARE/b2share/blob/520326cd8eccc5e1e14bebc60a4db0c7a50732e4/b2share/config.py#L373)
- "<RECORD_UUID>" can be observed in URL of the record/draft landing page.

Either construct the DOI within UI code or add the DOI information to `/api/records/<RECORD_UUID>` response.

### LINKED ISSUE #1743: Support newlines in record descriptions
By community request.

### LINKED ISSUE #1744: Add delete draft record button in draft edit form
A user should be able to delete an existing draft in the draft record edit form, so a button should be added with a confirmation request.

An API call already exists for this, so this should be fairly easy to add.

### LINKED ISSUE #1748: Adding small files from B2DROP in B2SHARE record form fails
From helpdesk #13086:

Adding files from B2DROP to B2SHARE record in B2SHARE interface.

PDF-files works fine. However, when adding a txt or json-file,
the file-upload stops with the message "Error: Internal Server
Error".

### LINKED ISSUE #1752: Allow record owners and administrators to see preview of draft record
By community request.

### LINKED ISSUE #1762: Changing search value doesn't reset page index in web UI
Screenshot 2020-02-25 at 17 30 26

When the new search result has less result pages than previous one and user is on a page higher than currently can be displayed, no results are shown and the pagination buttons are broken.

User then must click on page number again to see the results or manually fix the URL.

Proposed fix:
- When search value changes, set the current page to 1

### LINKED ISSUE #1773: License selector does not open in edit record form

### PR DESCRIPTION
**Pull request by hevp**
This PR implements the following issues:
- closes #1623 
- closes #1683 
- closes #1692 
- closes #1726 
- closes #1743 
- closes #1744
- closes #1748 
- closes #1752 
- closes #1762 
- closes #1773

### CODE REVIEW

File: webui/src/components/editrecord.jsx [hunk 1]
```diff
@@ -622,19 +648,31 @@ const EditRecord = React.createClass({
         const text = this.state.waitingForServer ? "Updating record, please wait..." :
                       this.isForPublication() ? 'Save and Publish' :
                       this.state.dirty ? 'Save Draft' : 'The draft is up to date';
+        const future_doi = this.props.record.get('b2share').get('future_doi') || "";
```
Comment (noblespirit) [inline webui/src/components/editrecord.jsx]: This causes a rendering bug if DOI creation is disabled.

Comment (hevp) [reply webui/src/components/editrecord.jsx]:
```suggestion
        const future_doi = this.props.record.get('b2share', Map()).get('future_doi', '') || "";
```

Comment (hevp) [reply webui/src/components/editrecord.jsx]: Previous commit updated

**Comment by noblespirit** [comment]
LGTM

**Pull request merged**